pages/project-details/view-details/[slug].js

```diff
@@ -0,0 +1,78 @@
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import Image from "next/image";
+import { useContext } from "react";
+import Link from "next/link";
+import { myViewsContext } from "@/pages/_app";
+
+export default function ViewDetails() {
+  const globalViews = useContext(myViewsContext);
+
+  const router = useRouter();
+  const { slug } = router.query;
+
+  const currentView = globalViews.find((view) => view.slug === slug);
+
+  if (!currentView) {
+    return <h1>404</h1>;
+  }
```

Comment:
if this case is entered, we can no longer navigate away from the page. can you give the user an option to escape your 404 case?

Reply:
added a Link